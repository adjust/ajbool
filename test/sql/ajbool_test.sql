BEGIN;
\pset null 'NULL'
SELECT true::ajbool, false::ajbool, NULL::bool::ajbool;
SELECT 't'::ajbool::bool, 'f'::ajbool::bool, 'u'::ajbool::bool;
CREATE TABLE test (i int, state ajbool);
INSERT INTO test
SELECT i, (CASE i%3 WHEN 0 THEN true WHEN 1 THEN false ELSE NULL END)::bool
FROM generate_series(1,10) i;
SELECT COUNT(1) FROM test where state;
SELECT COUNT(1) FROM test where not state;
ROLLBACK;