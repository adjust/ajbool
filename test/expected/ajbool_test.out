\pset null 'NULL'
SELECT true::ajbool, false::ajbool, NULL::bool::ajbool;
 ajbool | ajbool | ajbool 
--------+--------+--------
 t      | f      | u
(1 row)

SELECT 't'::ajbool::bool, 'f'::ajbool::bool, 'u'::ajbool::bool;
 bool | bool | bool 
------+------+------
 t    | f    | NULL
(1 row)

CREATE TABLE test (i int, state ajbool);
INSERT INTO test
SELECT i, (CASE i%3 WHEN 0 THEN true WHEN 1 THEN false ELSE NULL END)::bool
FROM generate_series(1,10) i;
SELECT COUNT(1) FROM test where state;
 count 
-------
     3
(1 row)

SELECT COUNT(1) FROM test where not state;
 count 
-------
     4
(1 row)

